benchmarking/frameworks/framework_base.py

```diff
@@ -62,6 +63,9 @@ def runBenchmark(self, info, benchmark, platform):
         platform_args = model["platform_args"] if "platform_args" in model \
             else {}
 
+        if (sys.version_info > (3, 0)) and 'timeout' in model.keys():
```

Comment:
just use "timeout" in model

I suspect you may want to check the test["timeout"] field as well, in case people have different timeouts in different tests.